src/mscorlib/src/System/String.Manipulation.cs

```diff
@@ -1160,7 +1162,7 @@ private String[] SplitInternal(ReadOnlySpan<char> separators, int count, StringS
                 return new String[] { this };
             }
 
-            int[] sepList = new int[Length];
+            Span<int> sepList = Length < StackallocStringLengthLimit ? stackalloc int[Length] : new int[Length];
```

Comment:
It would be nice to avoid the new allocation here every time even for larger Lengths. 

We may want to introduce internal ValueListBuilder<T> to help with it. It would be similar to https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/Text/ValueStringBuilder.cs, but it would be much simpler without all the string specific handling (e.g. it should just have a single Append method that takes T).

Reply:
should i create issue for it?

Reply:
It would be nice to fix this as part of  #6136 - the use of ArrayPool is mentioned there as one of ideas. It should also help with cases where your current fix makes things a bit slower.

Reply:
okay, i will take a look

Reply:
@jkotas what API for it do you expect? We can`t return array to pool while returning result as ValueStringBuilder does.
See https://gist.github.com/cod7alex/d1299959de8af5a18e9107c304ee1343

Reply:
Yes, the shape should be a bit different to make it work for this use case. E.g. there can be AsReadOnlySpan() method that returns the current list as `ReadOnlySpan<T>` and there should be `Dispose()` method that returns the array back to the bool if there is any.

Reply:
Also, if you think that it would be better to give this some other name than ValueListBuilder - feel free to change it.